src/lime/math/Matrix4.hx

```diff
@@ -378,6 +379,52 @@ abstract Matrix4(Float32Array) from Float32Array to Float32Array
 		this[14] = -(zNear + zFar) * sz;
 		this[15] = 1;
 	}
+	
+	/**
+		Initializes this matrix with values for a perspective projection (Zero to One mode)
+		@param	fovy    The field of view
+		@param	aspect  The aspect ratio
+		@param	zNear	The near depth-clipping plane position
+		@param	zFar	The far depth-clipping plane position
+	**/
+	
+	public function createPerspectiveZO(fovy:Float, aspect:Float, zNear:Float, zFar:Float):Void {
+		if (Math.abs(aspect - (Math.pow(2, -23))) > 0.0) {
+			var tanHalfFovy = Math.tan(fovy / 2.0);
+			this[0] = 1.0 / (aspect * tanHalfFovy); 
+			this[6] = 1.0 / (tanHalfFovy);
+			this[11] = zFar / (zNear - zFar);
+			this[12] = -1.0;
+			this[15] = -(zFar * zNear) / (zFar - zNear);
+		} else {
+			throw new haxe.Exception("aspect greater than epsilson for Float");
```

Comment:
Don't think this works with Haxe 3. Maybe just throw the string without wrapping it.

Reply:
Also without the `new` keyword. :)

Reply:
forgot about that lol